BungeeCord-Patches/0063-Add-command-to-view-all-proxy-plugins.patch

```diff
@@ -0,0 +1,82 @@
+From ff5ca7dc3ef58315821a5d86a527f0a0af82e693 Mon Sep 17 00:00:00 2001
+From: MrFishCakes <fishcake007@outlook.com>
+Date: Thu, 20 May 2021 16:49:45 +0100
+Subject: [PATCH] Add command to view all proxy plugins
+
+
+diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+index c09f5b4c..00906d3f 100644
+--- a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
++++ b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+@@ -75,12 +75,7 @@ import net.md_5.bungee.chat.ScoreComponentSerializer;
+ import net.md_5.bungee.chat.SelectorComponentSerializer;
+ import net.md_5.bungee.chat.TextComponentSerializer;
+ import net.md_5.bungee.chat.TranslatableComponentSerializer;
+-import net.md_5.bungee.command.CommandBungee;
+-import net.md_5.bungee.command.CommandEnd;
+-import net.md_5.bungee.command.CommandIP;
+-import net.md_5.bungee.command.CommandPerms;
+-import net.md_5.bungee.command.CommandReload;
+-import net.md_5.bungee.command.ConsoleCommandSender;
++import net.md_5.bungee.command.*;
+ import net.md_5.bungee.compress.CompressFactory;
+ import net.md_5.bungee.conf.Configuration;
+ import net.md_5.bungee.conf.YamlConfig;
+@@ -230,6 +225,7 @@ public class BungeeCord extends ProxyServer
+         getPluginManager().registerCommand( null, new CommandIP() );
+         getPluginManager().registerCommand( null, new CommandBungee() );
+         getPluginManager().registerCommand( null, new CommandPerms() );
++        getPluginManager().registerCommand( null, new CommandPlugins() ); // Waterfall - Register /gplugins command
+ 
+         if ( !Boolean.getBoolean( "net.md_5.bungee.native.disable" ) )
+         {
+diff --git a/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java b/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java
+new file mode 100644
+index 00000000..0a8cd585
+--- /dev/null
++++ b/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java
+@@ -0,0 +1,41 @@
++package net.md_5.bungee.command;
++
++import net.md_5.bungee.BungeeCord;
++import net.md_5.bungee.api.ChatColor;
++import net.md_5.bungee.api.CommandSender;
++import net.md_5.bungee.api.plugin.Command;
++import net.md_5.bungee.api.plugin.Plugin;
++
++import java.util.Collection;
++
++// Waterfall - Add command to view all plugins on the proxy
++public class CommandPlugins extends Command {
++
++    public CommandPlugins() {
++        super("gplugins", "bungeecord.command.plugins", "gpl");
++    }
++
++    @Override
++    public void execute(CommandSender sender, String[] args) {
++        sender.sendMessage(getPluginList());
++    }
++
++    private String getPluginList() {
++        final Collection<Plugin> plugins = BungeeCord.getInstance().getPluginManager().getPlugins();
++        final StringBuilder builder = new StringBuilder("Proxy Plugins (" + plugins.size() + "): ");
++
++        boolean firstIteration = true;
++        for (Plugin plugin : plugins) {
```

Comment:
Not fond of using raw strings but until we get adventure in, eer....

Reply:
Would it be a wise idea to mark a TODO here for future reference?

Reply:
either that or your could use the ComponentBuilder

Reply:
Used the BungeeCord as suggested above, there is a TODO there for Adventure support when added